### ISSUE TITLE
extra_sources not available in docker con

### ISSUE THREAD

**Issue by dgrisham** [testground/testground#1213]
#### Version information:

I'm running the most recent version of `testground` on `master`:

```
$ git log | head -n1
commit 3da7c61944ad273856c45d6d90a20eaf9e0dfdb6
```

Arch Linux, x64

#### Description:

Hey! I'm attempting to run tests in Docker using the `extra_sources` feature. I have the following lines in my manifest set up to include my local versions of two repositories as extra sources:

```
[extra_sources]
docker_go = [
    "../test-datasets",
    "../extra/go-bitswap",
    "../extra/go-peertaskqueue"
]
```

and I've tried a few corresponding replacement directives in the test plan's `go.mod`, such as

```
replace github.com/ipfs/go-bitswap => ./extra/go-bitswap
replace github.com/ipfs/go-peertaskqueue => ./extra/go-peertaskqueue
```

and

```
replace github.com/ipfs/go-bitswap => /extra/go-bitswap
replace github.com/ipfs/go-peertaskqueue => /extra/go-peertaskqueue
```

and

```
replace github.com/ipfs/go-bitswap => /go-bitswap
replace github.com/ipfs/go-peertaskqueue => /go-peertaskqueue
```

and

```
replace github.com/ipfs/go-bitswap => ./go-bitswap
replace github.com/ipfs/go-peertaskqueue => ./go-peertaskqueue
```

When I run the test using the `local:docker` runner and `docker:go` builder, I get the following error:

```
go: github.com/ipfs/go-bitswap@v0.2.20: parsing /extra/go-bitswap/go.mod: open /extra/go-bitswap/go.mod: no such file or directory
```

I've tried various combinations of `go.mod` replace directives, but nothing seems to work. I'm fairly certain that my extra sources directories are being found, because when I change them to invalid values I get an error when testground tries to read the directories.

Am I doing something wrong here, or is this feature not quite complete? I know it was only on the `oni` branch awhile back, but was under the impression that it had been merged into master.

Thanks!

**Comment by max-i-am** [comment]
I get the same issue when trying to add extra_sources to my test case. Inside the docker container the files are not available and probably not added at all. Therefore, I get the same error `../test-datasets: no such file or directory`.
Does somebody have a workaround or hot fix?

**Comment by nikosft** [comment]
The same here. Any progress on that?

**Comment by nikosft** [comment]
The problem seems to be that the `go mod download` command is executed before copying external sources. The following worked for me. Suppose the folder where your module source is located in named `mymodule`

- Copy `mymodule` folder inside your testplan  directory (where your go.mod file is located)
- Add the following [before this line](https://github.com/testground/testground/blob/adb40b17068ff63ed0005d0bbe0f284a60669a75/pkg/build/docker_go.go#L583)
```
COPY /plan/mymodule ${PLAN_DIR}/mymodule
```
- (Recomile testground)
-  Add the following line to you go.mod file:
```
replace github.com/<url of the module> => /plan/mymodule
```
You have to perform these steps only once.

**Comment by max-i-am** [comment]
My solution was to add a new section the "skip runtime image = true" to the manifest.toml.

```
[builders."docker:go"]
enabled = true
skip_runtime_image = true
```

I had to add only that feature and make it true and after that I was able to find the folder inside the docker containers. 
I detailed the same solution in the discussion forum [here](https://github.com/testground/testground/discussions/1244#discussioncomment-970441).

**Comment by nikosft** [comment]
I tried that as well. Indeed with this command I can see the folder but the folder is loaded after go mod download is executed. Hence, I couldn’t use it for adding module source code.
